partner_compassion/models/partner_compassion.py

```diff
@@ -215,6 +223,71 @@ def _compute_survey_input_count(self):
         for survey in self:
             survey.survey_input_count = len(survey.survey_inputs)
 
+    def _compute_criminal_record(self):
+        for partner in self:
+            partner.criminal_record = (
+                self.env["ir.attachment"].sudo()
+                    .search(
+                    [
+                        ("name", "like", "Criminal record"),
+                        ("res_id", "=", partner.id),
+                        ("res_model", "=", partner._name),
+                    ],
+                    limit=1,
+                )
+                .datas
+            )
+
+    def _inverse_criminal_record(self):
+        attachment_obj = self.env["ir.attachment"].sudo()
+        for partner in self:
+            criminal_record = partner.criminal_record
+            if criminal_record:
+                name = (
+                    "Criminal record "
+                    + partner.name
+                    + _get_file_type(criminal_record)
+                )
+                attachment_obj.create(
+                    {
+                        "datas_fname": name,
+                        "res_model": partner._name,
+                        "res_id": partner.id,
+                        "datas": criminal_record,
+                        "name": name,
+                    }
+                )
+                # search for all registrations and write completed task
+                registrations = self.env["event.registration"].sudo().search([
+                    ("partner_id", "=", partner.id)
+                ])
+                for registration in registrations:
+                    if registration:
+                        registration.write(
+                            {
+                                "completed_task_ids": [
+                                    (
+                                        4,
+                                        registration.env.ref(
+                                            "website_event_compassion.task_criminal"
+                                        ).id,
+                                    ),
+                                ]
+                            }
+                        )
```

Comment:
You cannot access event registrations from this module. You should inherit the method inside `website_event_compassion` to add this logic where you want.

Reply:
Corrected with commit ca8ec5c